src/Analyzers/CSharp/Analyzers/InlineDeclaration/CSharpInlineDeclarationDiagnosticAnalyzer.cs

```diff
@@ -181,6 +181,16 @@ private void AnalyzeSyntaxNode(SyntaxNodeAnalysisContext context, INamedTypeSymb
                 return;
             }
 
+            // If the local variable only has one declarator, then report the suggestion on the whole
+            // declaration.  Otherwise, return the suggestion only on the single declarator.
+            var reportNode = localDeclaration.Variables.Count == 1
+                ? (SyntaxNode)localDeclaration
+                : localDeclarator;
+
+            // Bail out early if the reportNode is outside the context's analysis span.
+            if (!context.ShouldAnalyzeSpan(reportNode.Span))
+                return;
```

Comment:
honestly, this seems like trying to be too precise.   seems like it woudl be more than fine to just do `if (!context.ShouldAnalyzeSpan(localDeclaration.Span))`.  It seems like only in rare cases would we just do a little extra work.  just my 2c.